internal/db/schema/migrations/oss/postgres/88/01_remove_session_connection_state.up.sql

```diff
@@ -0,0 +1,231 @@
+    as $$
+  begin
+    if new.termination_reason is not null then
+      perform  from
+        session
+      where
+        public_id = new.public_id and
+        public_id not in (
+            select session_id
+              from session_connection
+            where
+              -- open connections will have a connected_time_range with an upper bound of infinity
+              upper(connected_time_range) > now()
+        );
+      if not found then
+        raise 'session %s has open connections', new.public_id;
+      end if;
+      -- check to see if there's a terminated state already, before inserting a
+      -- new one.
+      perform from
+        session_state ss
+      where
+        ss.session_id = new.public_id and
+        ss.state = 'terminated';
+      if found then
+        return new;
+      end if;
+    insert into session_state (session_id, state)
+      values (new.public_id, 'terminated');
+    end if;
+    return new;
+  end;
+  $$ language plpgsql;
+
+  create trigger update_session_state_on_termination_reason after update of termination_reason on session
+    for each row execute procedure update_session_state_on_termination_reason();
+
+  create or replace function wh_insert_session_connection() returns trigger
+    as $$
+    declare
+  new_row wh_session_connection_accumulating_fact%rowtype;
+  begin
+    with
+      authorized_timestamp (date_dim_key, time_dim_key, ts) as (
+        select wh_date_key(create_time), wh_time_key(create_time), create_time
+        from session_connection
+        where public_id = new.public_id
+          and connected_time_range is null
+      ),
+      session_dimension (host_dim_key, user_dim_key, credential_group_dim_key) as (
+        select host_key, user_key, credential_group_key
+        from wh_session_accumulating_fact
+        where session_id = new.session_id
+      )
+    insert into wh_session_connection_accumulating_fact (
+        connection_id,
+        session_id,
+        host_key,
+        user_key,
+        credential_group_key,
+        connection_authorized_date_key,
+        connection_authorized_time_key,
+        connection_authorized_time,
+        client_tcp_address,
+        client_tcp_port_number,
+        endpoint_tcp_address,
+        endpoint_tcp_port_number,
+        bytes_up,
+        bytes_down
+    )
+    select new.public_id,
+       new.session_id,
+       session_dimension.host_dim_key,
+       session_dimension.user_dim_key,
+       session_dimension.credential_group_dim_key,
+       authorized_timestamp.date_dim_key,
+       authorized_timestamp.time_dim_key,
+       authorized_timestamp.ts,
+       new.client_tcp_address,
+       new.client_tcp_port,
+       new.endpoint_tcp_address,
+       new.endpoint_tcp_port,
+       new.bytes_up,
+       new.bytes_down
+    from authorized_timestamp,
+       session_dimension
+         returning * into strict new_row;
+    return null;
```

Comment:
Here, and elsewhere can you use [river formatting](https://github.com/hashicorp/boundary/blob/16d23fa55f6915b388f23c2dbc5a673249ba8823/sql-style-guide.md?plain=1#L130-L216), i.e.:

```suggestion
    with
      authorized_timestamp (date_dim_key, time_dim_key, ts) as (
        select wh_date_key(create_time), wh_time_key(create_time), create_time
          from session_connection
         where public_id = new.public_id
           and connected_time_range is null
    ),
      session_dimension (host_dim_key, user_dim_key, credential_group_dim_key) as (
        select host_key, user_key, credential_group_key
          from wh_session_accumulating_fact
         where session_id = new.session_id
    )
    insert into wh_session_connection_accumulating_fact (
                connection_id,
                session_id,
                host_key,
                user_key,
                credential_group_key,
                connection_authorized_date_key,
                connection_authorized_time_key,
                connection_authorized_time,
                client_tcp_address,
                client_tcp_port_number,
                endpoint_tcp_address,
                endpoint_tcp_port_number,
                bytes_up,
                bytes_down
    )
    select new.public_id,
           new.session_id,
           session_dimension.host_dim_key,
           session_dimension.user_dim_key,
           session_dimension.credential_group_dim_key,
           authorized_timestamp.date_dim_key,
           authorized_timestamp.time_dim_key,
           authorized_timestamp.ts,
           new.client_tcp_address,
           new.client_tcp_port,
           new.endpoint_tcp_address,
           new.endpoint_tcp_port,
           new.bytes_up,
           new.bytes_down
      from authorized_timestamp,
           session_dimension
 returning * into strict new_row;
 
    return null;
```